### PR TITLE
Allow opening multiple iframes simultaneously

### PR DESCRIPTION
**Pull request by lboyette-okta**
Resolves: OKTA-96993

@rchild-okta 
@ravikanodia-okta

### CODE REVIEW

File: test/spec/token.js [hunk 1]
```diff
@@ -112,6 +112,63 @@ define(function(require) {
...
+
+        // getWithoutPrompt, but don't resolve
+        var secondPrompt = context.client.token.getWithoutPrompt({
+          sessionToken: 'testSessionToken2',
+          state: oauthUtil.mockedState2,
+          nonce: oauthUtil.mockedNonce2
+        });
+
+        // assert that two iframes are open
+        expect(iframes.length).toBe(2);
```
Comment (rchild-okta) [inline test/spec/token.js]: Interesting - this works? I.e. I'm surprised that we can do a document.getElementsByTagName, which returns an array of elements, and that array of elements will dynamically update without having to search for the iframes again.

File: test/spec/token.js [hunk 2]
```diff
@@ -112,6 +112,63 @@ define(function(require) {
...
+
+        // resolve both prompts
+        context.emitter.emit('trigger', oauthUtil.mockedState);
+        context.emitter.emit('trigger', oauthUtil.mockedState2);
+
+        return Q.all([firstPrompt, secondPrompt])
+        .then(function() {
+          // make sure both iframes were destroyed
+          expect(iframes.length).toBe(0);
```
Comment (rchild-okta) [inline test/spec/token.js]: Should we also be checking that our tokens are what we expect here?

File: test/spec/token.js (hunk 1 above)
Comment (lboyette-okta) [inline test/spec/token.js]: `document.getElementsByTagName` returns an HTMLCollection, which is live.

> An HTMLCollection in the HTML DOM is live; it is automatically updated when the underlying document is changed.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection

File: test/spec/token.js (hunk 2 above)
Comment (lboyette-okta) [inline test/spec/token.js]: Sure

File: test/spec/token.js (hunk 1 above)
Comment (rchild-okta) [inline test/spec/token.js]: What?! That's kind of amazing. Okay, awesome.

**Pull request merged**